src/imports/register-extensions.js

```diff
@@ -53,9 +71,15 @@ const registerExtensions = (keyword, object, includeGlobal = false,
       const extensionPath = path.dirname(packageJsonPath);
       const extension = require(extensionPath); // eslint-disable-line global-require
 
-      extension[registerFunction](object, dependencyPath);
+      if (extension[registerFunction]) {
+        registeredExtension = extension[registerFunction](object, keyword);
+      } else {
+        logger.warn(`No register function found for extension ${dependency}`);
+      }
```

Comment:
This doesn't cover a case where registerFunction is truthy but not a function.  In the case of the function name being passed in by the parent of this class and then calling the function if it is truthy we should maybe check to make sure it is of type function?